benchmarks/prototype/blockwise_fp8_training/bench_triton_fp8_blockwise_act_quant_lhs.py

```diff
@@ -0,0 +1,224 @@
+    block_size = config.block_size
+
+    def verify_outputs(
+        y_naive: torch.Tensor,
+        s_naive: torch.Tensor,
+        y_triton: torch.Tensor,
+        s_triton: torch.Tensor,
+        rtol: float = 1e-2,
+        atol: float = 1e-2,
+    ):
+        """Verify that Triton and naive implementations produce similar results."""
+
+        # Convert FP8 back to float for comparison
+        y_naive_float = y_naive.to(torch.float32)
+        y_triton_float = y_triton.to(torch.float32)
+
+        # Check quantized values are close
+        try:
+            torch.testing.assert_close(
+                y_naive_float,
+                y_triton_float,
+                rtol=rtol,
+                atol=atol,
+                msg="Quantized values differ between naive and Triton implementations",
+            )
+        except AssertionError as e:
+            max_diff = (y_naive_float - y_triton_float).abs().max().item()
+            print(f"WARNING: Scales differ! Max diff: {max_diff}")
+            print(
+                f"  Naive scale range: [{y_naive_float.min():.6f}, {y_naive_float.max():.6f}]"
+            )
+            print(
+                f"  Triton scale range: [{y_triton_float.min():.6f}, {y_triton_float.max():.6f}]"
+            )
+            print(f"  Error details: {e}")
+
+        try:
+            torch.testing.assert_close(
+                s_naive,
+                s_triton,
+                rtol=rtol,
+                atol=atol,
+                msg="Scales differ between naive and Triton implementations",
+            )
+        except AssertionError as e:
+            max_diff = (s_naive - s_triton).abs().max().item()
+            print(f"WARNING: Scales differ! Max diff: {max_diff}")
+            print(f"  Naive scale range: [{s_naive.min():.6f}, {s_naive.max():.6f}]")
+            print(f"  Triton scale range: [{s_triton.min():.6f}, {s_triton.max():.6f}]")
+            print(f"  Error details: {e}")
+
+    input_tensor = torch.randn(
+        M,
+        K,
+        dtype=torch.bfloat16,
+        device=device,
+    )
+
+    # Benchmark naive implementation
+    # naive_impl_c = torch.compile(torch_blockwise_scale_act_quant_lhs)
+    y_naive, s_naive = torch_blockwise_scale_act_quant_lhs(input_tensor, block_size)
+    naive_time_us = benchmark_cuda_function_in_microseconds(
+        torch_blockwise_scale_act_quant_lhs,
+        input_tensor,
+        block_size,
+    )
+
+    # Benchmark Triton implementation
+    triton_impl_c = torch.compile(triton_fp8_blockwise_act_quant_lhs)
+    y_triton, s_triton = triton_impl_c(input_tensor, block_size)
+    triton_time_us = benchmark_cuda_function_in_microseconds(
+        triton_impl_c,
+        input_tensor,
+        block_size,
+    )
+
+    # Verify correctness (optional, can comment out for pure benchmarking)
+    verify_outputs(y_naive, s_naive, y_triton, s_triton)
+
+    # Memory bandwidth calculations
+    bytes_per_input_el = torch.finfo(torch.float32).bits / 8
```

Comment:
you're using bf16 input tensor, but calculating memory bandwidth using fp32 bytes per element.

to avoid this type of bug, just use `input_tensor.dtype` rather than hardcoding. same goes for fp8 outputs and scales.

i'd make this change across all the bench scripts